src/models/network/NetworkBaseController.js

```diff
@@ -111,48 +111,57 @@ export default class NetworkBaseController {
   }
 
   checkLocalContractsDeployed(throwIfFail = false) {
-    let msg;
+    this._handleErrorMessage(this._errorForLocalContractsDeployed(), throwIfFail);
+  }
+
+  _errorForLocalContractsDeployed() {
     const [contractsDeployed, contractsMissing] = _.partition(this.packageFile.contractAliases, (alias) => this.isContractDeployed(alias));
     const contractsChanged = _.filter(contractsDeployed, (alias) => this.hasContractChanged(alias));
 
     if (!_.isEmpty(contractsMissing)) {
-      msg = `Contracts ${contractsMissing.join(', ')} are not deployed.`;
+      return `Contracts ${contractsMissing.join(', ')} are not deployed.`;
     } else if (!_.isEmpty(contractsChanged)) {
-      msg = `Contracts ${contractsChanged.join(', ')} have changed since the last deploy.`;
+      return `Contracts ${contractsChanged.join(', ')} have changed since the last deploy.`;
     }
-
-    if (msg && throwIfFail) throw Error(msg);
-    else if (msg) log.info(msg);    
   }
 
   checkLocalContractDeployed(contractAlias, throwIfFail = false) {
-    let msg;
+    this._handleErrorMessage(this._errorForLocalContractDeployed(contractAlias), throwIfFail);
+  }
+
+  _errorForLocalContractDeployed(contractAlias) {
     if (!this.isContractDefined(contractAlias)) {
-      msg = `Contract ${contractAlias} not found in application or stdlib`;
+      return `Contract ${contractAlias} not found in this project`;
     } else if (!this.isContractDeployed(contractAlias)) {
-      msg = `Contract ${contractAlias} is not deployed to ${this.network}.`;
+      return `Contract ${contractAlias} is not deployed to ${this.network}.`;
     } else if (this.hasContractChanged(contractAlias)) {
-      msg = `Contract ${contractAlias} has changed locally since the last deploy, consider running 'zos push'.`;
+      return `Contract ${contractAlias} has changed locally since the last deploy, consider running 'zos push'.`;
     }
+  }
 
+  _handleErrorMessage(msg, throwIfFail = false) {
     if (msg && throwIfFail) throw Error(msg);
     else if (msg) log.info(msg);
   }
 
   hasContractChanged(contractAlias) {
-    if (!this.packageFile.hasContract(contractAlias)) return false;
+    if (!this.isLocalContract(contractAlias)) return false;
     if (!this.isContractDeployed(contractAlias)) return true;
     const contractName = this.packageFile.contract(contractAlias);
     const contractClass = Contracts.getFromLocal(contractName);
     return !this.networkFile.hasSameBytecode(contractAlias, contractClass)
   }
 
+  isLocalContract(contractAlias) {
+    return this.packageFile.hasContract(contractAlias)
+  }
+
   isContractDefined(contractAlias) {
     return this.packageFile.hasContract(contractAlias)
   }
 
   isContractDeployed(contractAlias) {
-    return !this.packageFile.hasContract(contractAlias) || this.networkFile.hasContract(contractAlias);
+    return !this.isLocalContract(contractAlias) || this.networkFile.hasContract(contractAlias);
```

Comment:
I'd move all these logic somewhere else, half of the lines of this Controller are related to contract deployment verifications

Reply:
Not sure if I agree. All verifications require access to both files, and are invoked by `script`s which typically interact with the network controller directly. Furthermore, logic is different between app and lib, which matches nicely with the hierarchy of base/app/lib controllers.

I can give it a shot by moving this to a ContractChecker or DeploymentVerifier, but I doubt it will make the design more clear, as the Controller will need to pass-through most calls to it.